edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest/KubernetesEnvironmentOperatorTest.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using FluentAssertions;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes.IntegrationTest.Client;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Xunit;
+
+    [Integration]
+    [Kubernetes]
+    public class KubernetesEnvironmentOperatorTest : IClassFixture<KubernetesClusterFixture>, IAsyncLifetime
```

Comment:
It's kind of complicated. The operator starts watching pod changes in k8s cluster and reacts to them. I wanted to follow through all pipeline of conversion from pod to module runtime info. That's why I'm getting data from KubernetesRuntimeInfoProvider and it looks like provider is the main character here.
Any ideas on how to make it more clear?

Reply:
Since an integration test tests the interaction between components more than the functionality of the individual components themselves, maybe the test class should be named to describe the _interaction_ that is being tested? But I wouldn't lose too much sleep over this...if the perfect name reveals itself later we can always update it.